src/name/ip_address.rs

```diff
@@ -14,6 +14,236 @@
+        Self::try_from_ascii(ip_address.as_bytes())
+    }
+
+    /// Constructs an `IpAddress` from this `IpAddressRef`
+    ///
+    /// Requires the `alloc` feature.
+    #[cfg(feature = "alloc")]
+    pub fn to_owned(&self) -> IpAddress {
+        match self {
+            IpAddressRef::IpV4AddressRef(ip_address, ip_address_octets) => IpAddress::IpV4Address(
+                String::from_utf8(ip_address.to_vec()).expect(VALID_IP_BY_CONSTRUCTION),
+                *ip_address_octets,
+            ),
+            IpAddressRef::IpV6AddressRef(ip_address, ip_address_octets) => IpAddress::IpV6Address(
+                String::from_utf8(ip_address.to_vec()).expect(VALID_IP_BY_CONSTRUCTION),
+                *ip_address_octets,
+            ),
+        }
+    }
+}
+
+#[cfg(feature = "std")]
+impl From<std::net::IpAddr> for IpAddress {
+    fn from(ip_address: std::net::IpAddr) -> IpAddress {
+        match ip_address {
+            std::net::IpAddr::V4(ip_address) => {
+                IpAddress::IpV4Address(ip_address.to_string(), ip_address.octets())
+            }
+            std::net::IpAddr::V6(ip_address) => {
+                IpAddress::IpV6Address(ip_address.to_string(), ip_address.octets())
+            }
+        }
+    }
+}
+
+impl<'a> From<IpAddressRef<'a>> for &'a str {
+    fn from(ip_address: IpAddressRef<'a>) -> &'a str {
+        match ip_address {
+            IpAddressRef::IpV4AddressRef(ip_address, _)
+            | IpAddressRef::IpV6AddressRef(ip_address, _) => {
+                core::str::from_utf8(ip_address).expect(VALID_IP_BY_CONSTRUCTION)
+            }
+        }
+    }
+}
+
+impl<'a> From<IpAddressRef<'a>> for &'a [u8] {
+    fn from(ip_address: IpAddressRef<'a>) -> &'a [u8] {
+        match ip_address {
+            IpAddressRef::IpV4AddressRef(ip_address, _)
+            | IpAddressRef::IpV6AddressRef(ip_address, _) => ip_address,
+        }
+    }
+}
+
+// https://tools.ietf.org/html/rfc5280#section-4.2.1.6 says:
+//   When the subjectAltName extension contains an iPAddress, the address
+//   MUST be stored in the octet string in "network byte order", as
+//   specified in [RFC791].  The least significant bit (LSB) of each octet
+//   is the LSB of the corresponding byte in the network address.  For IP
+//   version 4, as specified in [RFC791], the octet string MUST contain
+//   exactly four octets.  For IP version 6, as specified in
+//   [RFC2460], the octet string MUST contain exactly sixteen octets.
+pub(super) fn presented_id_matches_reference_id(
+    presented_id: untrusted::Input,
+    reference_id: untrusted::Input,
+) -> Result<bool, Error> {
+    if presented_id.len() != reference_id.len() {
+        return Ok(false);
+    }
+
+    let mut presented_ip_address = untrusted::Reader::new(presented_id);
+    let mut reference_ip_address = untrusted::Reader::new(reference_id);
+    loop {
+        let presented_ip_address_byte = presented_ip_address.read_byte().unwrap();
```

Comment:
So far you've merely asserted that the `presented_id` is the same length as the `reference_id`, so I don't think `unwrap()`ping here is safe? Since the comments mentions the expected lengths, I think we should check for those here as early as possible.

Reply:
You are absolutely right. Updating PR.

Reply:
I have pushed 7573333 which corrects this.

Reply:
Seems unfortunate that `untrusted` doesn't have some kind of iterator or `for_each()` support.